core/descriptors/src/org/jetbrains/kotlin/types/KotlinTypeFactory.kt

```diff
@@ -74,19 +74,30 @@ object KotlinTypeFactory {
         arguments: List<TypeProjection>,
         nullable: Boolean,
         kotlinTypeRefiner: KotlinTypeRefiner? = null
+    ) = simpleType(annotations.toDefaultAttributes(), constructor, arguments, nullable, kotlinTypeRefiner)
+
+    @JvmStatic
+    @JvmOverloads
+    @OptIn(TypeRefinement::class)
+    fun simpleType(
```

Comment:
For each type factory function, there are two versions: one with an attributes argument and one with an annotations argument. I've preserved the annotations argument version to avoid making small (potentially breaking) changes throughout the compiler as this function is very widely used.